### ISSUE TITLE
Heretic: A couple of issues

### ISSUE THREAD

**Issue by hfc2x** [Russian-Doom/russian-doom#188]
Greetings. I've been playing through Heretic today (latest build), and there are a couple things I found that I don't know if they have been addressed yet. They're very minor, however.

First, is the fact that the Green Key displays "Blue Key" when picked up, and second, that after completing E1M9, the map doesn't stay as marked in the overworld map.

I don't know if there are other minor issues like these still in the game, but I'll probably update this post if I find more.

**Comment by JNechaevsky** [comment]
Hello @hfc2x and thanks for your time, it's very appreciated! About issues:

1) Yeah, this is my bad, I have set wrong strings for keys. Will be fixed very soon.
2) This is probably vanilla behavior, I need to check and see what I can do.

**Comment by hfc2x** [comment]
>     2. This is probably vanilla behavior, I need to check and see what I can do.
It's not. I checked with both DOS v1.0 and 1.3, and secret maps remain marked upon completion.

By the way, this is completely unrelated, but I tried looking for an option for disabling completion screen on end-of-episode maps for both Doom and Heretic, but there isn't one. Not even launching the game with the `-vanilla` parameter disables it. Could this be made into a toggleable option? Sorry if it's too shameless of me to ask 😄

**Comment by JNechaevsky** [comment]
Just checked DOS version, situation is interesting:
* If you warp to `E1M9` and exit level normally, cross will remain on the world map.
* However, if you warp to `E1M6`, go to the secret exit, then finish `E1M9` cross will not appear.

It's same in RH. Fair enough, need to do something about it. :smiley:

As about disabling intermission screen, sure, I can disable it for `-vanilla`.
Please do not ask for in-game option, though, that's something should not happen. :astonished:

**Comment by JNechaevsky** [comment]
Fixed missing cross in https://github.com/JNechaevsky/russian-doom/commit/3027562da6e41938d0a098904f57213500a8cd8a, thanks! 

I've also noticed that shadows are not applied to the crossed while in "finished" state, and applied only in "now entering:" state. Definitely must be fixed.

**Comment by JNechaevsky** [comment]
Looks likes everything done here. However, if you have more suggestions or something, please feel free to explain. I'm thinking to release RH soon, a lot of improvements were made, including consolidation 4:3 and 16:9 into one executable and absolutely lovely automap rotate+overlay mode ([screenshot](https://user-images.githubusercontent.com/21193394/80471534-351f8780-894c-11ea-91af-ff72c00c9811.png)).

**Comment by hfc2x** [comment]
If Russian Heretic is getting all the features from Russian Doom, then I don't think I have anything to suggest. I haven't found any issues in my latest playthrough, other than what I mentioned, so I guess this issue can be closed 😄

EDIT: I think one thing that can be added is to remove the new, hardest difficulty setting from the New Game menu when playing with `-vanilla`, but still be able to play in that level if launched with `-skill 6`, or loaded from a previous save, so as to not break everything lol.

**Comment by hfc2x** [comment]
Oops, I think I closed this too soon lol.

Ending of Heretic Episode 3 doesn't scroll, it instead simply shows FINAL2 graphic (Heresiarch face) after the text screen, instead of showing FINAL1 (glass orb showing Corvus) and scrolling up to FINAL2.

**Comment by JNechaevsky** [comment]
Yeah, that's because I can't handle proper RAW screens scrolling in widescreen mode, always getting odd effects. The best I can do, though, is to add back scrolling in 4:3 mode, while showing just a static picture in 16:9. :/

**Comment by fabiangreffrath** [comment]
You could scroll a 4:3 picture in 16:9 mode with black bars to the left and right.

**Comment by JNechaevsky** [comment]
Oh, bars is a least problem here. The real problem is `p1` picture (`FINAL1`, orb), which is scrolling oddly ([screenshot](https://user-images.githubusercontent.com/21193394/80908371-07f61f00-8d28-11ea-8f9b-75c55bf08616.png)). Picture `p2` (`FINAL2`, demon), though, scrolling absolutely okay.

Here's the code:
```
void F_DemonScroll(void)
{
    byte *p1;
    byte *p2;
    static int yval = 0;
    static int nextscroll = 0;

    if (finalecount < nextscroll)
    {
        return;
    }
    p1 = W_CacheLumpName(DEH_String("FINAL1"), PU_LEVEL);
    p2 = W_CacheLumpName(DEH_String("FINAL2"), PU_LEVEL);

    if (finalecount < 70)
    {
        V_CopyScaledBuffer(I_VideoBuffer, p1, ORIGHEIGHT * ORIGWIDTH);
        nextscroll = finalecount;
        
        return;
    }
    if (yval < 64000)
    {
        V_CopyScaledBuffer(I_VideoBuffer, p2 + ORIGHEIGHT * ORIGWIDTH - yval, yval);
        V_CopyScaledBuffer(I_VideoBuffer + (yval << hires), p1, ORIGHEIGHT * ORIGWIDTH - yval);

        yval += SCREENWIDTH;
        nextscroll = finalecount + 3;
    }
    else
    {                           //else, we'll just sit here and wait, for now
        V_CopyScaledBuffer(I_VideoBuffer, p2, ORIGWIDTH * ORIGHEIGHT);
    }
}
```

**Comment by hfc2x** [comment]
Just out of curiosity: what happens when you provide 16:9 RAW screens when running in 16:9 mode? Do they render correctly, or do they look glitched as well?

**Comment by JNechaevsky** [comment]
It's a 16:9 Doom titlepic, converted into RAW. Ouch.

![изображение](https://user-images.githubusercontent.com/21193394/81328314-5ba89580-90a5-11ea-8a95-e7f3c66f953a.png)

**Comment by Dasperal** [comment]
Fixed in #249

**Comment by JNechaevsky** [comment]
Looks likes everything is fixed here. A new, highly improved is many aspects version of RHeretic will be labeled as version 5.0 and released after RD 5.0, which is, hopefully, will happen in the middle of September. It was a long time... 🙂
